### PR TITLE
[DISCOVERY-223] - adopt poetry & update README

### PR DESCRIPTION
**Pull request by bruno-fs**
Yet another PR related to DISCOVERY-223, normalizing quipucords development stack.

This adds proper support for poetry.

### CODE REVIEW

**Comment by abellotti** [comment]
❤️ this, waiting impatiently 👍

**Comment by abellotti** [comment]
Testing on Mac:

- poetry install ✅ 
- poetry shell ✅

File: Makefile [hunk 1]
```diff
@@ -49,17 +51,43 @@ clean-db:
...
+lock-build-requirements:
+	poetry run pip-compile $(PIP_COMPILE_ARGS) -r --resolver=backtracking --quiet --allow-unsafe --generate-hashes --output-file=requirements-build.txt requirements-build.in
+
+search-build-requirements:
+	cat requirements*.txt | grep -vE '(^ )|(#)' | awk '{print $$1}' | sed 's/==/ /' | \
+	xargs -P$(PARALLEL_NUM) -n2 poetry run pybuild-deps find-build-deps 2> /dev/null >> requirements-build.in || true
+
+update-requirements:
+	poetry update
+	$(MAKE) lock-requirements PIP_COMPILE_ARGS="--upgrade"
```
Comment (bruno-fs) [inline Makefile:81]: rules for managing `requirements.txt`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -27,9 +27,7 @@ branch should correspond to an associated issue opened on the main repository
 git clone https://github.com/quipucords/quipucords
 cd quipucords
 git checkout -b issues/my_issue_#
-pip install -r dev-requirements.txt
-docker-compose up -d
-docker-compose exec qpc-db psql -c 'alter role qpc with CREATEDB'
+poetry install
```
Comment (abellotti) [inline CONTRIBUTING.md:30]: should this be --with dev since the pip was with dev-requirements.txt ?

**Comment by abellotti** [comment]
@bruno-fs PR looks good, minor question above.

Also, why are those files changed in this PR ?  seems unrelated:

```
quipucords/api/insights_report/tests_insights_report.py
quipucords/quipucords/settings.py
quipucords/tests/integration/conftest.py
quipucords/tests/utils/container_wrappers.py
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (bruno-fs) [reply CONTRIBUTING.md:30]: I thought so, but then I realised `poetry install` also installs dev dependencies.

**Comment by bruno-fs** [comment]
> Also, are those files supposed to be in this PR ? they seems unrelated:
> 
> ```
> quipucords/api/insights_report/tests_insights_report.py
> quipucords/quipucords/settings.py
> quipucords/tests/integration/conftest.py
> quipucords/tests/utils/container_wrappers.py
> ```
The only change that isn't related is the one to `quipucords/api/insights_report/tests_insights_report.py`.
I did because the test is flaky on CI (due to the default timeout). I could just re-run the pipeline and it would pass, but I decided to just fix it - I can move to a separate PR with just this.


The other ones are loosely related - not to poetry but to the changes to Makefile + docs.

- The changes to `quipucords/quipucords/settings.py` were made to support the updated make directive for standing up a postgres container
- changes to `quipucords/tests/integration/conftest.py` and `quipucords/tests/utils/container_wrappers.py` were a bugfix(+error log improvement) for a bug introduced with the change to settings.

**Comment by abellotti** [comment]
This is fine to keep the tests_insights_report.py changes here since it's fixing the flaky tests.  Thanks for the explanation for the other 3.

File: CONTRIBUTING.md (hunk 2 above)
Comment (abellotti) [reply CONTRIBUTING.md:30]: Gotcha, It's just that you had `poetry install --with dev` in the .github/workflows/test.yml file so I thought it was necessary.

**Review by abellotti — APPROVED** [review]

**Comment by abellotti** [comment]
@bruno-fs PR looks great 👍  all good from my side on Mac. Thanks for writing this up so quickly!!

File: CONTRIBUTING.md (hunk 2 above)
Comment (bruno-fs) [reply CONTRIBUTING.md:30]: yeah... lemme remove that to avoid confusion

**Review by nicolearagao — APPROVED** [review]
Awesome work! LGTM 👍🏼

**Pull request merged**